src/stdlib_strings_format_string.fypp

```diff
@@ -0,0 +1,52 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+submodule (stdlib_strings) stdlib_strings_format_string
+
+    implicit none
+    integer, parameter :: buffer_len = 512
+
+contains
+
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure format_string_${type[0]}$${kind}$
+        !! Format ${type}$ variable as character sequence
+        character(len=buffer_len) :: buffer
+        integer :: stat
+        if(present(fmt)) then
+            write(buffer, fmt, iostat=stat) val
+            if(stat == 0) then
+                string = trim(adjustl(buffer))
```

Comment:
I would not left adjust the buffer but only trim trailing whitespace from the buffer:

For a calls like this I would expect that the width of the format is honored:

```
print '(a)', format_string(1, '(i0)'), format_string(1, '(i5)'), format_string(-1, '(i5)')
```

Resulting in an output like:
```
1
    1
   -1
```
While the current behaviour is more like
```
1
1
-1
```

Reply:
It is indeed better to remove `adjustl`, thank you~